dex/src/main/java/io/crate/data/CollectingBatchIterator.java

```diff
@@ -72,10 +86,9 @@ public static BatchIterator summingLong(BatchIterator source) {
         return new CloseAssertingBatchIterator(new CollectingBatchIterator<>(source, collector, numCols));
     }
 
-    private CollectingBatchIterator(BatchIterator source, Collector<Row, A, ? extends Iterable<Row>> collector, int numCols) {
-        this.source = source;
-        this.collector = collector;
-        this.rowData = new RowColumns(numCols);
+    public static <A> BatchIterator newInstance(BatchIterator source, Function<BatchIterator, CompletableFuture<? extends Iterable<Row>>> consumer,
```

Comment:
Generic type param is not used? Also, please re-format to have the second arg in an extra line